### DISCUSSION TITLE
how can I read the updated content in a custom node after [ updateNode ]

### DISCUSSION THREAD

**Discussion by lindaoling** [bcakmakoglu/vue-flow#1422] [Q&A] [answered]
a custom node like
```javascript
<template>
  <div>
    <slot name="label">custom node：{{ data.label }}</slot>
  </div>
</template>
<script setup>
defineProps({
  data: {
    type: Object,
    required: true,
  },
})
</script>
```
use it
```javascript
<VueFlow>
    <template #node-custom="nodeProps">
        <CustomNode :data="nodeProps" />
    </template>
</VueFlow>
```
update by updateNode
```javascript
updateNode('1',{label:'new label'})
```
it work when i use default node
but not work for custom node
how can I read the updated content in a custom node?

**Comment by bcakmakoglu** [comment, marked as answer]
You're not passing the data object as props, that's your mistake.
`nodeProps` is an object of type [`NodeProps`](https://vueflow.dev/typedocs/interfaces/NodeProps.html)

So if you wanna pass the data object as a prop you need to do `<CustomNode :data="nodeProps.data" />`.
